src/enzo/Grid_AddFeedbackSphere.C

```diff
@@ -922,10 +922,10 @@ int grid::AddFeedbackSphere(Star *cstar, int level, float radius, float DensityU
 	    }
 
 	    if (SNColourNum > 0)
-	      BaryonField[SNColourNum][index] *= factor;
+	      BaryonField[SNColourNum][index] = metallicity * BaryonField[DensNum][index];
```

Comment:
Won't this set the `SNColourNum` field to the same value as the `MetalNum` field? Is that the right thing to do?

Reply:
Good catch. They should be the same fields. In principle, it should happen for all of the metal fields. I'll expand it to include them.